src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/course/restapi/CoursesControllerTest.kt

```diff
@@ -163,31 +174,32 @@ class CoursesControllerTest(
   @Nested
   inner class GetOfferingsTests {
     @Test
-    fun `get all offerings for a course`() {
-      val courseId = repository.allCourses().first().id!!
-      every { coursesService.offeringsForCourse(courseId) } returns repository.offeringsForCourse(courseId)
+    fun `coursesCourseIdOfferingsGet with JWT returns 200 with correct body`() {
+      val offerings = listOf(
+        OfferingEntityFactory().withOrganisationId("OF1").withContactEmail("of1@digital.justice.gov.uk").produce(),
```

Comment:
Is Offering(organisationId = "OF1", contactEmail = "of1@digital.justice.gov.uk") simpler? Clearer?

Reply:
This is a good point. On the one hand, I could make the case for Factories being useful for generating each of our entity objects because they may become quite complex as time goes on, or that it's worth using them if, for instance, `Referral` was a base class with multiple different implementations. On the other, if these Factories are otherwise only working as static wrappers around simpler classes, then it's an anti-pattern and doesn't actually help to reduce complexity at all.

I'm happy to remove them if you think YAGNI applies in this situation?

Reply:
Spoke with @Gweaton about this, and have decided to merge as-is, for two reasons - dxw regularly use factories; and in, this case, they're very useful for setting universal placeholder data when testing.

Reply:
I agree with Peter that not using factories is simpler and clearer in this situation, but I think the pros of using factories outweigh the cons of this specific implementation being a bit more complex.

Using factories will allow us to create reusable entities with shared traits if we need them in future, and will help reduce some some boilerplate code. We use them in the frontend codebase too, so if, in future, we have team members writing both UI and API code, there'll be some consistency across the two sides in terms of design choices.